xla/service/gpu/runtime/collectives.cc

```diff
@@ -179,20 +181,36 @@ absl::Status AsyncDoneImpl(const ServiceExecutableRunOptions* run_options,
 #endif  // XLA_ENABLE_XCCL
 }
 
+// TODO: shall we use GpuDriver::LaunchKernel() to avoid macros here ?
 #if XLA_ENABLE_XCCL
```

Comment:
Here it'd be good to make sure the `gpu_stream`, and variables below are wrapped in a `#if GOOGLE_CUDA || TENSORFLOW_USE_ROCM` that way they aren't marked as unused. maybe have the `#if XLA_ENABLE_XCCL` also require either `GOOGLE_CUDA` or `TENSORFLOW_USE_ROCM`?

Reply:
There were a few additional issues as well, I tried to fix them. For example we cannot use a cc_library for a cuda kernel, probably tf_gpu_kernel_library is the right choice (but I cannot verify whether it will work for ROCM). I will try to get this merged.

Reply:
Turns out that tf_gpu_kernel/tsl_gpu_kernel also didn't work out, so I am now fixing it by creating 2 different libraries for cuda and rocm (i.e. for cuda we keep using cuda_library, for rocm we will be using cc_library like suggested in this PR).

Reply:
Strange, I specifically checked that cuda kernel really builds by adding errors to 'mock_nccl_call' and compiling it.
Actually, cuda_library() is just cc_library with 'cuda_default_opts':
```
def cuda_library(copts = [], **kwargs):
    """Wrapper over cc_library which adds default CUDA options."""
    native.cc_library(copts = cuda_default_copts() + copts, **kwargs)
```
However, there is indeed an error in this commit: cc_library 'collectives' does not set library-specific macros, i.e.:
```
local_defines = if_cuda_is_configured(["GOOGLE_CUDA=1"]) + if_rocm_is_configured(["TENSORFLOW_USE_ROCM=1"]),
```
otherwise 'collectives' is compiled silently without GPU support

Reply:
> Here it'd be good to make sure the `gpu_stream`, and variables below are wrapped in a `#if GOOGLE_CUDA || TENSORFLOW_USE_ROCM` that way they aren't marked as unused. maybe have the `#if XLA_ENABLE_XCCL` also require either `GOOGLE_CUDA` or `TENSORFLOW_USE_ROCM`?

Yes, this is a good point, we can wrap all local variables with a macro guard and the parameter 'gpu_stream' with [[maybe_unused]]